components/FlexibleHeader/tests/unit/FlexibleHeaderShiftedOffscreenTests.m

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialFlexibleHeader.h"
+
+@interface FlexibleHeaderShiftedOffscreenTests : XCTestCase
+@end
+
+@implementation FlexibleHeaderShiftedOffscreenTests
+
+#pragma mark - Programmatic shifting using -shiftHeader(On|Off)ScreenAnimated:
+
+#pragma mark Without a tracking scroll view
+
+- (void)testDefaultIsFalseWithoutTrackingScrollView {
+  // Given
+  MDCFlexibleHeaderView *fhvc = [[MDCFlexibleHeaderView alloc] init];
+
+  // Then
+  XCTAssertFalse(fhvc.shiftedOffscreen);
+  XCTAssertEqualWithAccuracy(CGRectGetMinY(fhvc.frame), 0, 0.001);
+}
+
+- (void)testBecomesTrueWhenShiftedOffscreenNotAnimatedWithoutTrackingScrollView {
+  // Given
+  MDCFlexibleHeaderView *fhvc = [[MDCFlexibleHeaderView alloc] init];
+
+  // When
+  [fhvc shiftHeaderOffScreenAnimated:NO];
+
+  // Then
+  XCTAssertTrue(fhvc.shiftedOffscreen);
+  XCTAssertEqualWithAccuracy(CGRectGetMinY(fhvc.frame), -fhvc.maximumHeight, 0.001);
+}
+
+- (void)testBecomesTrueWhenShiftedOffscreenAnimatedWithoutTrackingScrollView {
```

Comment:
Shouldn't this test wait until the animation finishes? Or should the test be named something like "becomesTrueBeforeAnimationWhen..."?

Reply:
Renamed the test.